src/figure/formation_layout.c

```diff
@@ -42,3 +42,14 @@ int formation_layout_position_y(int layout, int index)
 {
     return FORMATION_LAYOUT_POSITION_Y[layout][index];
 }
+
+void formation_layout_position(int layout, int index, formation *m, uint8_t *x, uint8_t *y)
```

Comment:
Using `formation *m` and calling it with a `const formation *` variable gives compiler warnings. Since you don't update the formation, declare the parameter as `const`.

Naming: a formation has multiple coordinates (current, home, standard, destination). When reading code that's calling this function, it's not clear that this function is using the destination, rather than one of the other coordinates. Rename the function name to make that explicit, for example: `formation_layout_get_destination`.

This function is only called from the animal figure code. You mentioned that we might need it for soldiers/enemies as well, but where `formation_layout_position_x/y` are currently used, they're not using destination so we cannot reuse this function anyway. I suggest moving it to animal.c as static function.

Reply:
I missed these comments!

I try my best but I never think about stuff like this.